tests/test_projectmetadata.py

```diff
@@ -126,6 +126,28 @@ def test_deprecation_warning_repo_of_type_package(tmp_path):
         Project(tmp_path, autostd=False)
 
 
+def test_deprecation_warning_requires_field(tmp_path):
+    # Verify that a deprecation warning is raised when the requires field is set
+    with pytest.warns(
+        ProjectConfigurationWarning,
+        match=re.escape(
+            "The requires field of the project.yml file is deprecated. Add dependencies/constraints"
+            " in the requirements.txt file of the project. The requires field will be dropped in the"
+            " next major release."
+        ),
+    ):
+        with (tmp_path / "project.yml").open("w") as fh:
+            fh.write(
+                """
+    name: testproject
+    downloadpath: libs
+    requires:
+      - std
+    """
```

Comment:
Ha, today I learned that leading whitespaces are ignored in yaml e.g. these are equivalent [no leading spaces](https://yaml-online-parser.appspot.com/?yaml=%0A-+just%3A+write+some%0A-+yaml%3A%0A++-+%5Bhere%2C+and%5D%0A++-+%7Bit%3A+updates%2C+in%3A+real-time%7D%0A&type=json) and [leading spaces](https://yaml-online-parser.appspot.com/?yaml=+++-+just%3A+write+some%0A+++-+yaml%3A+%0A+++++-+%5Bhere%2C+and%5D%0A+++++-+%7Bit%3A+updates%2C+in%3A+real-time%7D%0A&type=json)